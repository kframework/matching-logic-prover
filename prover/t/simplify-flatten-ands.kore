symbol f(Int) : Int
axiom a: \equals(f(1), 1)
claim \equals(f(\and(1, \and())), 1)
strategy simplify.flatten-ands
       . apply-equation -> a at 0 by []
       . reflexivity
