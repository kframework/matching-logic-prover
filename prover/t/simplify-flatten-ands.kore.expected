<prover>
  <exit-code>
    0
  </exit-code>
  <goals>
    <goal>
      id: ., parent: .
      <claim>
        \and ( .Patterns )
      </claim>
      <k>
        .
      </k>n<expected>
        .
      </expected>
      <local-context>
        .LocalDeclCellSet
      </local-context>
    <trace>
      .
    </trace>
  </goals>
  <declarations>
    <declaration>
      axiom a : \equals ( f ( 1 , .Patterns ) , 1 )
    </declaration> <declaration>
      axiom ax0 : \equals ( f ( \and ( 1 , \and ( .Patterns ) , .Patterns ) , .Patterns ) , 1 )
    </declaration> <declaration>
      symbol f ( Int , .Sorts ) : Int
    </declaration>
  </declarations>
</prover>
